packages/origin-247-certificate/src/offchain-certificate/offchain-certificate.service.ts

```diff
@@ -75,35 +76,50 @@ export class OffchainCertificateService<T = null> {
         return certs;
     }
 
-    public async batchClaim(command: IClaimCommand[]): Promise<void> {
-        await Promise.all(
-            command.map(async (c: IClaimCommand) => {
-                await this.claim(c);
-            })
-        );
+    public async batchClaim(commands: IClaimCommand[]): Promise<void> {
+        for (const command of commands) {
+            await this.claim(command);
+        }
     }
 
-    public async batchTransfer(command: ITransferCommand[]): Promise<void> {
-        await Promise.all(
-            command.map(async (c: ITransferCommand) => {
-                await this.transfer(c);
-            })
-        );
+    public async batchTransfer(commands: ITransferCommand[]): Promise<void> {
+        for (const command of commands) {
+            await this.transfer(command);
+        }
     }
 
-    // TODO: add actual validation
-    private validateCommand(command: CertificateCommand): void {}
+    private async createAggregate(event: ICertificateEvent): Promise<CertificateAggregate> {
+        const readModel = await this.readModelRepo.getByInternalCertificateId(
```

Comment:
never ever create aggregate from read model. It's just prohibited. Read model is not guaranteed to be up to date, and here we need as fresh model as possible.